src/Illuminate/Foundation/Auth/Access/AuthorizesRequests.php

```diff
@@ -48,7 +48,7 @@ public function authorizeForUser($user, $ability, $arguments = [])
      */
     protected function parseAbilityAndArguments($ability, $arguments)
     {
-        if (is_string($ability)) {
+        if (is_string($ability) && ! is_subclass_of($ability, 'Illuminate\Database\Eloquent\Model')) {
```

Comment:
it's not a restriction rather then an extension.  most probably nobody will call any of his ability to 'App\Post' or something like that. not to mention that a '\' in the ability name is rather ambiguous. so if the ability name is something like this 'App\Post' ie a fully qualified class name the we assume it's a "Automatically Determining Policy Methods" from https://laravel.com/docs/master/authorization